TaskSQL/Queries.sql

```diff
@@ -0,0 +1,296 @@
+	
+	PRIMARY KEY(CardNumber),
+	
+	FOREIGN KEY (AccountId) REFERENCES Accounts(Id)
+);
+
+--Filling out the table "Banks"
+INSERT INTO Banks(BankName) 
+values ('Bank1'),
+	('Alfa-bank'), 
+	('Belarusbank'), 
+	('BPSBank'), 
+	('BelAgroPromBank');
+
+--Filling out the table "Cities"
+INSERT INTO Cities(City) 
+values ('Gomel'), 
+	('Minsk'), 
+	('Grodno'), 
+	('Mogilev'), 
+	('Brest'), 
+	('Vitebsk');
+
+--Filling out the table "BanksCities"
+INSERT INTO BanksCities (BankId, CityId) 
+values ('1','2'), 
+	('1','1'), ('1','3'), 
+	('1','5'), ('2','2'), 
+	('2','3'), ('2','5'), 
+	('2','1'), ('2','3'), 
+	('3','5'), ('3','3'), 
+	('3','1'), ('4','2'), 
+	('4','4'), ('5','6'), 
+	('2','6'), ('5','1');
+
+--Filling out the table "SocialStatus"
+INSERT INTO SocialStatus(SocialStatus) 
+values ('Pensioner'), 
+	('Disabled person'), 
+	('And other'),
+	('Some_new_status');
+
+--Filling out the table "Owners"
+INSERT INTO Owners(SocialStatusId, OwnerName) 
+values ('2', 'Ivasenko V.V.'), 
+	('3', 'Ermolenko A.A.'), 
+	('1', 'Ivanov I.I.'), 
+	('3', 'Bokitko A.S.'), 
+	('1', 'Pechenkin A.A.');
+
+--Filling out the table "Accounts"
+INSERT INTO Accounts(Balance, BankId, OwnerId) 
+values ('200.30', '1', '1'), 
+	('300.21', '1', '1'), 
+	('2015', '3', '2'), 
+	('545', '2', '3'), 
+	('4125', '4', '5'), 
+	('2158', '5', '4');
+
+--Filling out the table "Cards"
+INSERT INTO Cards(AccountId, CardNumber, CardBalance) 
+values ('6', '1234567891025684', '60.21'), 
+	('5', '6589423682354', '125.03'), 
+	('4', '5454886452213', '235.06'), 
+	('3', '54848303255452', '1225'), 
+	('2', '458787933656006', '300.21'), 
+	('1', '8759232656552', '125.50'), 
+	('6', '87592326565', '200.30');
+
+--Item 1: Output the name of banks that are in the city X
+SELECT BankName 
+FROM Banks 
+	INNER JOIN BanksCities ON Banks.Id = BanksCities.BankId 
+	INNER JOIN Cities ON Cities.Id = BanksCities.CityId 
+WHERE Cities.City LIKE 'Gomel';
+
+--Item 2: Get a list of cards indicating the name of the owner, balance and the name of the bank
+SELECT CardNumber, CardBalance, bankName, OwnerName 
+FROM Cards 
+	INNER JOIN Accounts ON Cards.AccountId = Accounts.Id 
+	INNER JOIN Banks ON Accounts.BankId = Banks.Id 
+	INNER JOIN Owners ON Accounts.OwnerId = Owners.Id;
+
+--Item 3: Show a list of bank accounts whose balance does not coincide with the sum of balance on cards. In a separate column, output the difference
+SELECT DISTINCT Accounts.Id,
+	MAX(Accounts.Balance) 'AccountBalance', 
+	SUM(Cards.CardBalance) 'SumOdCardBalanceWithOneAccount', 
+	ABS(SUM(Cards.CardBalance) - MAX(Accounts.Balance)) 'Difference'
+FROM Accounts 
+	INNER JOIN Cards ON Accounts.Id = Cards.AccountId 
+GROUP BY (Accounts.Id)
+HAVING SUM(Cards.CardBalance) <> MAX(Accounts.Balance)
+
+--Item 4: Output count bank cards for everyone social status with GROUP BY
+SELECT SocialStatus, COUNT (Cards.CardNumber) 'CountCards'  
+FROM SocialStatus 
+	FULL OUTER JOIN Owners ON SocialStatus.Id = Owners.SocialStatusId

```

Comment:
We need to use other JOIN here to avoid records like [NULL, 13] for owners without socialStatus